src/Components/Server/src/Circuits/RenderBatchWriter.cs

```diff
@@ -130,16 +130,15 @@ int Write(in ArrayRange<RenderTreeFrame> frames)
 
         void Write(in RenderTreeFrame frame)
```

Comment:
Unfortunately, for server-side Blazor, this means reference frames now take 20 bytes each instead of 16 before, and in most of the frame types, a large part of those 20 bytes are just padding.

We could reduce it to 14 bytes per frame by:

* Writing the `FrameType` as a 2-byte value instead of 4 bytes. It's a `short` anyway. This would save 2 bytes.
* Having a separate `FrameType` value for attribute frames that have a nonempty event handler ID. Then for that new frame type, we wouldn't need to write out the `AttributeValue` (which is always left blank in this case). This would let all the possible frame type bodies fit into 12 bytes, and hence saves 4 bytes.

However I'm not doing this now because it can be done non-breakingly in the future if we want.

Reply:
Filed https://github.com/aspnet/AspNetCore/issues/12306 and put it right on the backlog

Reply:
Sounds good! I was going to suggest to make the frame type a  byte directly as I wouldn't expect us to have more than 256 frame types, but I realize that would disrupt the alignment and possibly slow things down.